src/main/java/gov/usgs/locator/LocMain.java

```diff
@@ -208,106 +391,196 @@ public static void main(String[] args) {
+
+            String line = "";
+            // each file is assumed to be a single message
+            while ((line = inputReader.readLine()) != null) {
+              requestString += line;
+            }
+          } catch (FileNotFoundException e) {
+            LOGGER.severe(e.toString());
+          } catch (IOException e) {
+            LOGGER.severe(e.toString());
+          } finally {
+            try {
+              if (inputReader != null) {
+                inputReader.close();
+              }
+            } catch (IOException e) {
+              LOGGER.severe(e.toString());
+            }
+          }
 
-    // set all handlers to the same formatter
-    for (Handler handler : rootLogger.getHandlers()) {
-      handler.setFormatter(new SimpleLogFormatter());
-    }
-  }
+          // parse into request
+          try {
+            request = new LocationRequest(Utility.fromJSONString(requestString));
+          } catch (Exception e) {
+            // parse failure
+            LOGGER.severe("Exception: " + e.toString());
 
-  /**
-   * This function converts a log level string into a logger level. This function converts a couple
-   * of non-standard logging levels / abbreviations.
-   *
-   * @param logLevel A String holding the desired log level
-   * @return A Level object containing the desired log level.
-   */
-  private Level getLogLevel(String logLevel) {
-    if (logLevel == null) {
-      return null;
-    }
-    try {
-      return Level.parse(logLevel.toUpperCase());
-    } catch (IllegalArgumentException e) {
-      if (logLevel.equalsIgnoreCase("DEBUG")) {
-        return Level.FINE;
-      }
-      if (logLevel.equalsIgnoreCase("WARN")) {
-        return Level.WARNING;
-      }
-      throw new IllegalArgumentException(
-          "Unresolved log level " + logLevel + " for java.util.logging", e);
-    }
-  }
+            StringWriter sw = new StringWriter();
+            PrintWriter pw = new PrintWriter(sw);
+            e.printStackTrace(pw);
+            LOGGER.severe("Trace: " + sw.toString());
 
-  /**
-   * This function extracts the file name from a given file path.
-   *
-   * @param filePath A String containing the full path to the file
-   * @return A String containing the file name extracted from the full path.
-   */
-  public static String getFileName(String filePath) {
-    // get the file name from input file path
-    int start = filePath.lastIndexOf("/");
-    if (start < 0) {
-      start = filePath.lastIndexOf("\\");
+            LOGGER.severe("File: " + fileName);
+            return (1);
+          }
+        } else {
+          // run as LocInput/LocOutput to get access to read/write routines
+          LocInput hydraIn = new LocInput();
+          if (!hydraIn.readHydra(inputPath + File.separatorChar + fileName)) {
+            LOGGER.severe(
+                "Failed to read hydra file: " + inputPath + File.separatorChar + fileName);
+            return (0);
```

Comment:
Will this keep the errored out file in the current directory, so the app will continue to try reading it and continue getting hung up on it? Below you check for archive to move file or delete if processed, maybe there should be an error dir too? Or did I miss some other handling in the code?

Reply:
It should be fine given how the function reads files from the directory, but added code to rename bad input files .error to be safe.